File: modules/dashboards/SystemHealth.py
# SDOA v1.2 compliant — Streamlit Dashboard
from base import Dashboard
import streamlit as st

class SystemHealth(Dashboard):
    MANIFEST = {
        "id": "SystemHealth",
        "runtime": "Python",
        "version": "1.2.0",  
        "dependencies": [
            "ProvisioningService", 
            "BunInstaller", 
            "LlmPolicyEngine"
        ]
    }

    def render(self):
        st.title("🛡️ SDOA Control Plane")
        
        # Sync with the Registry state
        policy = self.registry.get("LlmPolicyEngine").getPolicy()
        
        col1, col2 = st.columns(2)
        with col1:
            st.metric("Primary LLM", policy['primary']['provider'].upper())
            if st.button("Force Economic Fail-over"):
                self.registry.get("LlmPolicyEngine").updatePolicy({"primary": {"provider": "ollama"}})
                st.rerun()

        with col2:
            status = "Healthy" if self.registry.get("ProvisioningService").verify_environment() else "Broken"
            st.metric("Runtime Status", status) 

File: modules/dashboards/SystemHealth.rs
// SDOA v1.2 compliant — Native Shell Dashboard
use sdoa_sdk::prelude::*;
use tauri::Manager;

#[derive(SdoaDashboard)]
#[sdoa(
    id = "SystemHealth",
    runtime = "Rust/Tauri",
    version = "2.0.0",
    dependencies = ["ProvisioningService", "LlmPolicyEngine"]
)]
pub struct SystemHealth;

impl SystemHealth {
    // This Rust method triggers the native Tauri window
    pub fn spawn_window(&self, app_handle: tauri::AppHandle) {
        let _window = tauri::WindowBuilder::new(
            &app_handle,
            "system_health",
            tauri::WindowUrl::App("health_view.html".into())
        )
        .title("SDOA System Control")
        .inner_size(1200.0, 800.0)
        .build()
        .unwrap();
        
        self.bump_patch("Migrated to native Tauri shell");
    }
}